include/aos/iam/identhandler.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_IDENTHANDLER_HPP_
+#define AOS_IDENTHANDLER_HPP_
+
+#include "aos/common/types.hpp"
+#include "aos/iam/config.hpp"
+
+namespace aos {
+namespace iam {
+namespace identhandler {
+
+/** @addtogroup iam Identification and Access Manager
+ *  @{
+ */
+
+/**
+ * Subjects changed observer interfaceю
+ */
+class SubjectsChangedObserverItf {
+public:
+    /**
+     * Subjects Changed Observer interface.
+     *
+     * @param[in] messages subject changed messages.
+     * @returns Error.
+     */
+    virtual Error SubjectsChanged(const Array<StaticString<cSubjectIDLen>>& messages) = 0;
+
+    /**
+     * Destroys subjects changed observer interface.
+     */
+    virtual ~SubjectsChangedObserverItf() = default;
+};
+
+/**
+ * Ident handler interface.
+ */
+class IdentHandlerItf {
+public:
+    /**
+     * Returns System ID.
+     *
+     * @param[out] systemdID result system ID.
+     * @returns Error.
+     */
+    virtual Error GetSystemID(String& systemdID) = 0;
```

Comment:
For small string we are usually use RetWithError<SystemID> instead of return in parameter list